cmd/snap-update-ns/update_test.go

```diff
@@ -20,16 +20,302 @@
 package main_test
 
 import (
+	"bytes"
+
+	. "gopkg.in/check.v1"
+
 	update "github.com/snapcore/snapd/cmd/snap-update-ns"
+	"github.com/snapcore/snapd/logger"
 	"github.com/snapcore/snapd/osutil"
+	"github.com/snapcore/snapd/testutil"
 )
 
-// testProfileUpdateContext implements MountProfileUpdate and is suitable for testing.
+type updateSuite struct {
+	testutil.BaseTest
+	log *bytes.Buffer
+}
+
+var _ = Suite(&updateSuite{})
+
+func (s *updateSuite) SetUpTest(c *C) {
+	s.BaseTest.SetUpTest(c)
+	buf, restore := logger.MockLogger()
+	s.BaseTest.AddCleanup(restore)
+	s.log = buf
+}
+
+func (s *updateSuite) TestSmoke(c *C) {
+	ctx := &testProfileUpdateContext{}
+	c.Assert(update.ExecuteMountProfileUpdate(ctx), IsNil)
+}
+
+func (s *updateSuite) TestUpdateFlow(c *C) {
+	// The flow of update is as follows:
+	// - the current profile and the desired profiles are loaded
+	// - the needed changes are computed
+	// - the needed changes are performed (one by one)
+	// - the updated current profile is saved
+	var loadedCurrent, loadedDesired, changesComputed, savedCurrent bool
+	var changesPerformed int
+	ctx := &testProfileUpdateContext{
+		loadCurrentProfile: func() (*osutil.MountProfile, error) {
+			loadedCurrent = true
+			return &osutil.MountProfile{}, nil
+		},
+		loadDesiredProfile: func() (*osutil.MountProfile, error) {
+			loadedDesired = true
+			return &osutil.MountProfile{}, nil
+		},
+		neededChanges: func(old, new *osutil.MountProfile) []*update.Change {
+			changesComputed = true
+			return []*update.Change{{}, {}}
+		},
+		performChange: func(change *update.Change, as *update.Assumptions) ([]*update.Change, error) {
+			changesPerformed++
```

Comment:
Could record the order in which changes are to be applied.

Reply:
Yeah, good idea. I'll make that so.

Reply:
Done